### ISSUE TITLE
Exact dataset labels

### ISSUE THREAD

**Issue by yoctta** [yinanhe/ForgeryNet#17]
Thank you for releasing the dataset, it is really valuable for deepfake researches.  Could you release the labels of testset and the exact manipulation methods of 16cls_labels?

**Comment by yinanhe** [comment]
1: FaceShifter
2: FS-GAN
3: DeepFakes
4: BlendFace
5: MMReplacement
6: DeepFakes-StarGAN-Stack
7: Talking Head Video
8: ATVG-Net
9: StarGAN-BlendFace-Stack
10: First Order Motion
11: StyleGAN2
12: MaskGAN
13: StarGAN2
14: SC-FEGAN
15: DiscoFaceGAN

The label of test set will be public soon.

**Comment by inglis92** [comment]
> 1: FaceShifter 2: FS-GAN 3: DeepFakes 4: BlendFace 5: MMReplacement 6: DeepFakes-StarGAN-Stack 7: Talking Head Video 8: ATVG-Net 9: StarGAN-BlendFace-Stack 10: First Order Motion 11: StyleGAN2 12: MaskGAN 13: StarGAN2 14: SC-FEGAN 15: DiscoFaceGAN
> 
> The label of test set will be public soon.

Hi, where can i find the label of test set? Have them already been released?

**Comment by LightningChan** [comment]
> 1: FaceShifter 2: FS-GAN 3: DeepFakes 4: BlendFace 5: MMReplacement 6: DeepFakes-StarGAN-Stack 7: Talking Head Video 8: ATVG-Net 9: StarGAN-BlendFace-Stack 10: First Order Motion 11: StyleGAN2 12: MaskGAN 13: StarGAN2 14: SC-FEGAN 15: DiscoFaceGAN
> 
> The label of test set will be public soon.

What are labels of 16~19？

**Comment by yinanhe** [comment]
> > 1: FaceShifter 2: FS-GAN 3: DeepFakes 4: BlendFace 5: MMReplacement 6: DeepFakes-StarGAN-Stack 7: Talking Head Video 8: ATVG-Net 9: StarGAN-BlendFace-Stack 10: First Order Motion 11: StyleGAN2 12: MaskGAN 13: StarGAN2 14: SC-FEGAN 15: DiscoFaceGAN
> > The label of test set will be public soon.
> 
> What are labels of 16~19？

real image

**Comment by LightningChan** [comment]
You mentioned four real sources in the paper. Actually, I would like to know which source each number corresponds to. Thx!

**Comment by yinanhe** [comment]
16: RAVDESS
17:  CREMA-D
18: AVSpeech
19: vox

**Comment by davide-coccomini** [comment]
Are the test set labels available now?

**Comment by LightningChan** [comment]
> 1: FaceShifter 2: FS-GAN 3: DeepFakes 4: BlendFace 5: MMReplacement 6: DeepFakes-StarGAN-Stack 7: Talking Head Video 8: ATVG-Net 9: StarGAN-BlendFace-Stack 10: First Order Motion 11: StyleGAN2 12: MaskGAN 13: StarGAN2 14: SC-FEGAN 15: DiscoFaceGAN
> 
> The label of test set will be public soon.

In third row in the file 'image_list.txt':
train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15
The 16cls_label '15' is different from category '7'.  So I should use which one to determine corresponding method? thx~

**Comment by yinanhe** [comment]
> > 1: FaceShifter 2: FS-GAN 3: DeepFakes 4: BlendFace 5: MMReplacement 6: DeepFakes-StarGAN-Stack 7: Talking Head Video 8: ATVG-Net 9: StarGAN-BlendFace-Stack 10: First Order Motion 11: StyleGAN2 12: MaskGAN 13: StarGAN2 14: SC-FEGAN 15: DiscoFaceGAN
> > The label of test set will be public soon.
> 
> In third row in the file 'image_list.txt': train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15 The 16cls_label '15' is different from category '7'. So I should use which one to determine corresponding method? thx~

7

**Comment by YoshihikoFuruhashi** [comment]
Allow me to confirm the above conclusions.

- `train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15` for example
  - 7
    - the subdirectory number == the method index? 🤔 
  - 15
    - What do the n-way numbers in `image_list.txt` mean? Can this be ignored?🤔 
  
---

### Counting results for subdirectory numbers and labels in the `image_list.txt`

Is this relationship between **sub-dir** and **method** correct? 🤔 

```bash
sub-dir two-way three-way n-way count method
1 1 2 8 99687 FaceShifter
2 1 1 7 199993 FS-GAN
3 1 2 3 100000 DeepFakes
4 1 2 2 100000 BlendFace
5 1 2 1 100000 MMReplacement
6 1 2 4 100000 DeepFakes-StarGAN-Stack
7 1 1 15 21570 Talking Head Video < train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15
8 1 1 10 96880 ATVG-Net
9 1 1 12 53947 StarGAN-BlendFace-Stack
10 1 1 14 29467 First Order Motion
11 1 1 13 29570 StyleGAN2
12 1 2 9 14785 MaskGAN
13 1 2 6 99949 StarGAN2
14 1 1 11 58444 SC-FEGAN
15 1 2 5 88089 DiscoFaceGAN
16 0 0 0 57047 RAVDESS
17 0 0 0 73736 CREMA-D
18 0 0 0 553264 AVSpeech
19 0 0 0 474877 VoxCeleb2
```

**Comment by yinanhe** [comment]
Yes, thank you for your example, this explanation is completely correct!

**Comment by Damon-Salvatore-liu** [comment]
> Allow me to confirm the above conclusions.
> 
> * `train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15` for example
>   
>   * 7
>     
>     * the subdirectory number == the method index? 🤔
>   * 15
>     
>     * What do the n-way numbers in `image_list.txt` mean? Can this be ignored?🤔
> 
> ### Counting results for subdirectory numbers and labels in the `image_list.txt`
> Is this relationship between **sub-dir** and **method** correct? 🤔
> 
> ```shell
> sub-dir two-way three-way n-way count method
> 1 1 2 8 99687 FaceShifter
> 2 1 1 7 199993 FS-GAN
> 3 1 2 3 100000 DeepFakes
> 4 1 2 2 100000 BlendFace
> 5 1 2 1 100000 MMReplacement
> 6 1 2 4 100000 DeepFakes-StarGAN-Stack
> 7 1 1 15 21570 Talking Head Video < train_release/7/2ac43aa43bf473f9a9c09b4b608619d3/b039e05887ddaad1a72964d5d1ab88d3/frame00038.jpg 1 1 15
> 8 1 1 10 96880 ATVG-Net
> 9 1 1 12 53947 StarGAN-BlendFace-Stack
> 10 1 1 14 29467 First Order Motion
> 11 1 1 13 29570 StyleGAN2
> 12 1 2 9 14785 MaskGAN
> 13 1 2 6 99949 StarGAN2
> 14 1 1 11 58444 SC-FEGAN
> 15 1 2 5 88089 DiscoFaceGAN
> 16 0 0 0 57047 RAVDESS
> 17 0 0 0 73736 CREMA-D
> 18 0 0 0 553264 AVSpeech
> 19 0 0 0 474877 VoxCeleb2
> ```

@YoshihikoFuruhashi Hello, could you tell me the difference between n-way and sub-dir

**Comment by YoshihikoFuruhashi** [comment]
@Damon-Salvatore-liu 

- sub-dir: The directory where the image data is stored.
- n-way: The label number of the method.

Use sub-dir == n-way, and ignore image_list.txt  as it contains errors.

@yinanhe Is this correct?

**Comment by alvaroqsaldanha** [comment]
@yinanhe Could you provide an answer to the last post? Is each subdirectory in train_release/ corresponding to the used method?

**Comment by alvaroqsaldanha** [comment]
@yinanhe Hello. Could you confirm what method is used in each subdir for image classification? Is ./1 FaceShifter or ATVG-Net?  
Thank you.
